### PR TITLE
[MDS-5714] AMS applicant information and Orgbook validation

### PR DESCRIPTION
**Pull request by henryoforeh-dev**
## Objective 

[MDS-5714](https://bcmines.atlassian.net/browse/MDS-5714)

- Implemented RenderOrgBookSearch component.
- Migration script to update project_summary, party, mine_party_appt_type_code and party_orgbook_entity tables.
- Implemented AMS applicant information.

image

### CODE REVIEW

File: services/core-api/app/api/parties/response_models.py [hunk 1]
```diff
@@ -135,7 +136,8 @@ def format(self, value):
         'now_party_appt': fields.Nested(NOW_PARTY_APPOINTMENT),
         'organization_guid': fields.String,
         'organization': fields.Nested(ORGANIZATION, skip_none=True),
-        "digital_wallet_connection_status": fields.String
+        "digital_wallet_connection_status": fields.String,
+        'middle_name': fields.String
```
Comment (matbusby-fw) [inline services/core-api/app/api/parties/response_models.py:140]: Is this right?  `middle_name` on the `PARTY_BUSINESS_ROLE_APPT` feels like a strange fit.

Comment (henryoforeh-dev) [reply services/core-api/app/api/parties/response_models.py:140]: This is the full screen of the code. It is actually on the Party reponse model which has first_name attribute.  On the front end, this is for Individual party type in AMS Applicant information.
image

File: services/common/src/components/forms/RenderOrgBookSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
+import React, { FC, useEffect, useRef, useState } from "react";
```
Comment (matbusby-fw) [inline services/common/src/components/forms/RenderOrgBookSearch.tsx:1]: Nice enhancement to this!

**Review by matbusby-fw — APPROVED** [review]
Nice work!

File: services/core-api/app/api/parties/response_models.py (hunk 1 above)
Comment (matbusby-fw) [reply services/core-api/app/api/parties/response_models.py:140]: Ah, okay!

**Review by asinn134 — APPROVED** [review]
Great work 👍

**Pull request merged**